controllers/machinedeletion_controller.go

```diff
@@ -63,19 +73,54 @@ func (r *MachineDeletionReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	if remediation = r.getRemediation(ctx, req); remediation == nil {
 		return ctrl.Result{}, nil
 	}
-	//not a machine based remediation
-	machineOwnerRef := getMachineOwnerRef(remediation)
-	if machineOwnerRef == nil {
-		return ctrl.Result{}, nil
+
+	var machine *unstructured.Unstructured
+	//Health check was done by NHC
+	if node, err := r.getNodeFromMdr(remediation); err == nil {
+		if machine, err = r.buildMachineFromNode(node); err != nil {
+			return ctrl.Result{}, err
+		}
+		if isMachineBelongToMasterNode(machine) {
+			return ctrl.Result{}, nil
+		}
+
+	} else { //Failed both in fetching the machine and in fetching the node
+		return ctrl.Result{}, err
 	}
-	//delete the machine
-	if err := r.deleteMachine(ctx, buildMachine(machineOwnerRef, remediation)); err != nil {
+
+	if err := r.deleteMachineOfNode(ctx, machine, remediation.Name); err != nil {
 		return ctrl.Result{}, err
 	}
 
 	return ctrl.Result{}, nil
 }
 
+func (r *MachineDeletionReconciler) deleteMachineOfNode(ctx context.Context, machine *unstructured.Unstructured, nodeName string) error {
+	//delete the machine
+	_ = r.Client.Delete(ctx, machine)
+
+	key := client.ObjectKey{
+		Name:      machine.GetName(),
+		Namespace: machine.GetNamespace(),
+	}
+
+	//verify machine is deleted
+	if err := r.Get(context.TODO(), key, machine); !errors.IsNotFound(err) {
```

Comment:
this is not really required if you check the `err` of `Delete()`

Reply:
I've added this verification after a conversation with Andrew.
The rational is that even if the previous delete does not return an error we are not guaranteed that machine is indeed deleted.
An example for that is when a finalizer is set on the machine - you can see it in one of the Unit Tests.

Reply:
this doesn't sound right.
you need to better explain the motivation. let's say there's a finalizer on the machine - what's expect from MDR to do with that?

Reply:
so we want to reconcile this indefinitely and write endless messages to the log in case the finalizer was never removed?

Reply:
This is the scenario as I understand it:
1. Delete was successful.
2. The machine was not deleted. 

Finalizer is one use-case which will satisfy those terms, but I don't know if it's the only one.

- I'm pretty sure that logging makes sense here.
- Regarding a possible indefinite reconcile that's a good point.

I can think of two arguments to support the chance of risking indefinite reconciling.
1.  Although the problem might be due to a Finalizer, it can also be something else that can be fixed. 
2. Even if the problem is Finalizer I don't think it means necessarily indefinite reconcile, It is possible that the finalizer will be removed over time - whether automatically or manually.

Reply:
> Finalizer is one use-case which will satisfy those terms, but I don't know if it's the only one.

There's no other use-case I'm familiar with as well, and I don't think there's one. unless we can prove other scenario, finalizer i s the only thing we may want to handle.


> Although the problem might be due to a Finalizer, it can also be something else that can be fixed.
I disagree with that. Unless there's a proven or at least good suspect that something like that can happen it doesn't make sense to assume this.

There's an API and the API promises something - if `Delete` request completed succesfully, you're guaranteed that the object will be deleted as soon as all finalizers will be removed. 
If you're starting to assume that the API is not working properly and violating its promises, you'll end up no where, as I could say this on any api-request you make.

Reply:
also, I've never seen such pattern in any other operator, can you point to some other operator that does that?
we're definitely not the only operator that deletes resources.

Reply:
>There's an API and the API promises something - if Delete request completed succesfully, you're guaranteed that the object will be deleted as soon as all finalizers will be removed.

I wasn't aware of that - I agree it's a strong argument against another reconcile.

>also, I've never seen such pattern in any other operator, can you point to some other operator that does that?
we're definitely not the only operator that deletes resources.

I'm not aware of anything specific.

So assuming we accept the premise that the only use case  for this behavior is due to a Finalizer:
- it make sense not return an error result.
- We can either
  - not write any log (in this case also remove the Get after deletion)
  - write log ( (in this case the Get after deletion is required in order to identify this situation)
 
 Between writing log/not writing log I think that writing log is the better option here, so there will be some indication that the remediation is at best pending or stuck at worst.